### PR TITLE
Update Nilearn API _check_same_fov

### LINKED ISSUE #872: nilearn 0.10.3  breaks nimare
```
     9 import pymare
     10 from joblib import Memory
---> 11 from nilearn._utils.niimg_conversions import _check_same_fov
     12 from nilearn.image import concat_imgs, resample_to_img
     13 from nilearn.input_data import NiftiMasker

ImportError: cannot import name '_check_same_fov' from 'nilearn._utils.niimg_conversions' (/usr/local/lib/python3.10/dist-packages/nilearn/_utils/niimg_conversions.py)
```

### PR DESCRIPTION
**Pull request by adelavega**
Fixes #872

### CODE REVIEW

**Comment by adelavega** [comment]
Looks like nilearn also changed how they handle `cmap` in `plot_roi`. 

Specifically, when we pass a custom cmap, on an image with only 1 level `[0, 1]`, it fails.

File: nimare/meta/ibma.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import pandas as pd
 import pymare
 from joblib import Memory
-from nilearn._utils.niimg_conversions import _check_same_fov
+from nilearn._utils.niimg_conversions import check_same_fov
```
Comment (jdkent) [inline nimare/meta/ibma.py:12]: if we want to continue to support 0.10.1 nilearn
```suggestion
try:
    # nilearn>=0.10.3
     from nilearn._utils.niimg_conversions import check_same_fov
except ImportError:
    # nilearn < 0.10.3
    from nilearn._utils.niimg_conversions import _check_same_fov as check_same_fov
```

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -232,6 +232,8 @@ jobs:
       - name: Download artifacts
         uses: actions/download-artifact@v2
       - name: Upload to CodeCov
-        uses: codecov/codecov-action@v2
+        uses: codecov/codecov-action@v3
         with:
           fail_ci_if_error: true
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (jdkent) [inline .github/workflows/testing.yml:239]: will see if this breaks when a pull request is opened from a fork instead of the original repo.

**Review by jdkent — APPROVED** [review]
Looks like this works now! thanks @remi-gau and @adelavega!

**Pull request merged**